sphinxcontrib/needs/directives/needfilter.py

```diff
@@ -65,7 +65,7 @@ def run(self):
             'show_filters': 'show_filters' in self.options,
             'show_legend': 'show_legend' in self.options,
             'layout': self.options.get("layout", "list"),
-            'export_id': self.options.get("export_id", ""),
+            'export_id': self.options.get("export_id"),
```

Comment:
would be better if the filter config were a class, rather than a dict. I won't touch that in this pull request, i'll just revert the change, since you're right, i don't know what the downstream code will do.

certainly `None` would make more sense than an empty string, semantically. right?